x-pack/legacy/plugins/reporting/server/routes/lib/authorized_user_pre_routing.ts

```diff
@@ -4,52 +4,44 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import Boom from 'boom';
-import { Legacy } from 'kibana';
+import { RequestHandler, RouteMethod } from 'src/core/server';
 import { AuthenticatedUser } from '../../../../../../plugins/security/server';
-import { ReportingConfig } from '../../../server';
-import { LevelLogger as Logger } from '../../../server/lib';
-import { ReportingSetupDeps } from '../../../server/types';
 import { getUserFactory } from '../../lib/get_user';
+import { ReportingInternalSetup, ReportingCore } from '../../core';
 
 const superuserRole = 'superuser';
 
-export type PreRoutingFunction = (
-  request: Legacy.Request
-) => Promise<Boom<null> | AuthenticatedUser | null>;
+export type RequestHandlerUser = RequestHandler extends (...a: infer U) => infer R
+  ? (user: AuthenticatedUser, ...a: U) => R
+  : never;
 
 export const authorizedUserPreRoutingFactory = function authorizedUserPreRoutingFn(
-  config: ReportingConfig,
-  plugins: ReportingSetupDeps,
-  logger: Logger
+  core: ReportingCore,
+  setupDeps: ReportingInternalSetup
 ) {
-  const getUser = getUserFactory(plugins.security, logger);
-  const { info: xpackInfo } = plugins.__LEGACY.plugins.xpack_main;
-
-  return async function authorizedUserPreRouting(request: Legacy.Request) {
-    if (!xpackInfo || !xpackInfo.isAvailable()) {
-      logger.warn('Unable to authorize user before xpack info is available.', [
-        'authorizedUserPreRouting',
-      ]);
-      return Boom.notFound();
-    }
-
-    const security = xpackInfo.feature('security');
-    if (!security.isEnabled() || !security.isAvailable()) {
-      return null;
-    }
-
-    const user = await getUser(request);
-
-    if (!user) {
-      return Boom.unauthorized(`Sorry, you aren't authenticated`);
-    }
-
-    const authorizedRoles = [superuserRole, ...(config.get('roles', 'allow') as string[])];
-    if (!user.roles.find((role) => authorizedRoles.includes(role))) {
-      return Boom.forbidden(`Sorry, you don't have access to Reporting`);
-    }
-
-    return user;
+  const config = core.getConfig();
+  const getUser = getUserFactory(setupDeps.security);
+
+  return <P, Q, B>(handler: RequestHandlerUser): RequestHandler<P, Q, B, RouteMethod> => {
+    return (context, req, res) => {
+      const user = getUser(req);
```

Comment:
I think we need to test if security is enabled, higher up here. Reporting should still work if security is no enabled.

Reply:
I'm working on that in https://github.com/elastic/kibana/pull/67233 since it affects the current tests